File: src/main/java/land/leets/domain/interview/presentation/dto/req/FixedInterviewRequest.java
package land.leets.domain.interview.presentation.dto.req;

import jakarta.validation.constraints.NotNull;
import land.leets.domain.interview.domain.Interview;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
public class FixedInterviewRequest {
    @NotNull
    private LocalDateTime fixedInterviewDate;

    @NotNull
    private String place;

    public void updateInterview(Interview interview) {
        interview.setFixedInterviewDate(this.fixedInterviewDate);
        interview.setPlace(this.place);
    }
}

File: src/main/kotlin/land/leets/domain/interview/presentation/dto/req/FixedInterviewRequest.kt
package land.leets.domain.interview.presentation.dto.req

import jakarta.validation.constraints.NotNull
import land.leets.domain.interview.domain.Interview
import java.time.LocalDateTime

data class FixedInterviewRequest(
    @field:NotNull
    val fixedInterviewDate: LocalDateTime,

    @field:NotNull
    val place: String
) {
    fun updateInterview(interview: Interview) {
        interview.fixedInterviewDate = this.fixedInterviewDate
        interview.place = this.place
    }
}
